### PR TITLE
EGL_MESA_drm_image: fix a couple typos and add a cursor use bit

### PR DESCRIPTION
**Pull request by 1ace**
This pulls in the Mesa changes from Kristian Høgsberg (@krh) and Nicolas Kaiser (@nikai3d)

https://cgit.freedesktop.org/mesa/mesa/commit/?id=e5169e9615e8391ea369415b356168717b8f7be0
https://cgit.freedesktop.org/mesa/mesa/commit/?id=ae5776c41f12515bb73c07ee2a0aed56cdd1a1ef

### CODE REVIEW

File: extensions/MESA/EGL_MESA_drm_image.txt [hunk 1]
```diff
@@ -66,6 +66,7 @@ New Tokens
 
         EGL_DRM_BUFFER_USE_SCANOUT_MESA		0x0001
         EGL_DRM_BUFFER_USE_SHARE_MESA		0x0002
+        EGL_DRM_BUFFER_USE_CURSOR_MESA	0x0004
```
Comment (oddhack) [inline extensions/MESA/EGL_MESA_drm_image.txt]: We normally name bitfield enums 'EGL_name_BIT[_vendor]', but that ship has probably sailed with this extension. However, the new bit does need to be added to egl.xml as part of this PR.

Comment (1ace) [reply extensions/MESA/EGL_MESA_drm_image.txt]: You're right, I completely forgot that.
I should let you regenerate the headers though, right?

Comment (oddhack) [reply extensions/MESA/EGL_MESA_drm_image.txt]: Yes, I'll regenerate them when this is accepted. Thanks!

File: api/egl.xml [hunk 2]
```diff
@@ -149,6 +149,7 @@
     <enums namespace="EGLDRMBufferUseMESAMask" type="bitmask" comment="EGL_DRM_BUFFER_USE_MESA bits">
         <enum value="0x00000001" name="EGL_DRM_BUFFER_USE_SCANOUT_MESA"/>
         <enum value="0x00000002" name="EGL_DRM_BUFFER_USE_SHARE_MESA"/>
+        <enum value="0x00000004" name="EGL_DRM_BUFFER_USE_CURSOR_MESA"/>
```
Comment (oddhack) [inline api/egl.xml]: This needed to be added to the \<extension> block as well - I did this and regenerated the header after merging this.

Comment (1ace) [reply api/egl.xml]: Indeed; should've grepped. Hopefully now I'll get it right in one go next time :)
Thanks for merging all my PR :+1:

**Pull request merged**